code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,181 @@
+	..()
+
+	if(!check_rights(R_EVENT))
+		return
+
+	if(href_list["remove"])
+		qdel(src)
+
+/datum/virology_goal/property_symptom
+	name = "Symptom With Properties Viral Sample Request"
+	var/goal_symptom //Type path of the symptom
+	var/goal_symptom_name
+	var/goal_property
+	var/goal_property_value
+
+/datum/virology_goal/property_symptom/New()
+	do
+		var/type = pick(subtypesof(/datum/symptom))
+		var/datum/symptom/S = new type()
+		goal_symptom = S.type
+		goal_symptom_name = S.name
+		goal_property = pick("resistance", "stealth", "stage rate", "transmittable")  
+		goal_property_value = rand(-18, 11)
+		switch(goal_property)
+			if("resistance")
+				goal_property_value += S.resistance
+			if("stealth")
+				goal_property_value += S.stealth
+			if("stage rate")
+				goal_property_value += S.stage_speed
+			if("transmittable")
+				goal_property_value += S.transmittable
+		qdel(S)
+	while(check_for_duplicate())
+
+/datum/virology_goal/property_symptom/check_for_duplicate()
+	. = FALSE
+	if(!goal_symptom || !goal_property || !goal_property_value)
+		return
+	for(var/datum/virology_goal/property_symptom/V in (GLOB.archived_virology_goals + GLOB.virology_goals))
+		if(goal_symptom == V.goal_symptom && goal_property == V.goal_property && goal_property_value == V.goal_property_value)
+			return TRUE
+
+/datum/virology_goal/property_symptom/get_report()
+	return {"<b>Effects of [goal_symptom_name] symptom and level [goal_property_value] [goal_property]</b><br>
+	Viral samples with a specific symptom and properties are required to study the effects of this symptom in various conditions. We need you to deliver [delivery_goal]u of viral samples containing the [goal_symptom_name] symptom and with the [goal_property] property at level [goal_property_value] along with 3 other symptoms to us through the cargo shuttle."}
+
+/datum/virology_goal/property_symptom/get_ui_report()
+	return {"Viral samples with a specific symptom and properties are required to study the effects of this symptom in various conditions. We need you to deliver [delivery_goal]u of viral samples containing the [goal_symptom_name] symptom and with the [goal_property] property at level [goal_property_value] along with 3 other symptoms to us through the cargo shuttle."}
+
+/datum/virology_goal/property_symptom/check_completion(list/datum/reagent/reagent_list)
+	. = FALSE
+	var/datum/reagent/blood/BL = locate() in reagent_list
+	if(!BL)
+		return
+	if(BL && BL.data && BL.data["viruses"])
+		for(var/datum/disease/advance/D in BL.data["viruses"])
+			if(length(D.symptoms) < 4) //We want 3 other symptoms alongside the requested one
+				continue
+			var/properties = D.GenerateProperties()
+			var/property = properties[goal_property]
+			if(property != goal_property_value)
+				continue
+			for(var/datum/symptom/S in D.symptoms)
+				if(!goal_symptom)
+					return
+				if(S.type != goal_symptom)
+					continue
+				delivered_amount = clamp(delivered_amount + BL.volume, 0, delivery_goal)
```

Comment:
Any particular reason to clamp it?

Reply:
the clamp is there so the ui doesnt show like 30 delivered out of 15 but now i realize that probably doesnt matter